streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -2686,7 +2686,28 @@ object ZStreamSpec extends ZIOBaseSpec {
                 _     <- f.join
               } yield assertTrue(count == 0)
             }
-          } @@ TestAspect.jvmOnly @@ nonFlaky(20)
+          } @@ TestAspect.jvmOnly @@ nonFlaky(20),
+          test("accumulates parallel errors") {
+            sealed abstract class DbError extends Product with Serializable
+            case object Missing           extends DbError
+            case object QtyTooLarge       extends DbError
+
+            for {
+              exit <- ZStream(1 to 2: _*)
+                        .mapZIOPar(3) {
+                          case 1 => ZIO.fail(Missing)
+                          case 2 => ZIO.fail(QtyTooLarge)
+                          case _ => ZIO.succeed(true)
+                        }
+                        .runDrain
+                        .exit
+            } yield assert(exit)(
+              failsCause(
+                containsCause[DbError](Cause.fail(Missing)) &&
+                  containsCause[DbError](Cause.fail(QtyTooLarge))
+              )
+            )
+          } @@ flaky
```

Comment:
Out of curiosity, what makes this test flaky?

Reply:
sometimes the runtime does not start the second fiber before the error signal lands in the writer